model/src/main/java/uk/gov/hmcts/payment/api/service/UserAwareDelegatingCardPaymentService.java

```diff
@@ -180,12 +188,19 @@ private static Predicate getPredicate(Root<Payment> root, CriteriaBuilder cb, Da
             predicates.add(cb.equal(paymentJoin.get("paymentMethod"), PaymentMethod.paymentMethodWith().name(paymentMethod).build()));
         }
 
+
+        // date_trunc not supported in hsqldb
+        Expression<Date> dateUpdatedExpr = isTestMode == true ?
+            cb.function("trunc", Date.class, paymentJoin.get("dateUpdated")) :
+            cb.function("date_trunc", Date.class, cb.literal("seconds"), paymentJoin.get("dateUpdated"));
```

Comment:
should be driven by an environment variable. i.e value for test will be different to prod one. but code shouldn't be different based on test